datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -1084,14 +1087,16 @@ private void deleteTopic(Datastream datastream) {
    * any instance after old unused tasks are cleaned up.
    */
   private void handleLeaderDoAssignment(boolean cleanUpOrphanNodes) {
-    boolean succeeded = true;
+    AtomicBoolean succeeded = new AtomicBoolean(true);
     List<String> liveInstances = Collections.emptyList();
     Map<String, Set<DatastreamTask>> previousAssignmentByInstance = Collections.emptyMap();
     Map<String, List<DatastreamTask>> newAssignmentsByInstance = Collections.emptyMap();
 
     try {
       List<DatastreamGroup> datastreamGroups = fetchDatastreamGroups();
-
+      List<DatastreamGroup> stoppingDatastreamGroups =
+          fetchDatastreamGroupsWithStatus(Collections.singletonList(DatastreamStatus.STOPPING));
+      _log.debug("stopping datastreams: {}", stoppingDatastreamGroups);
```

Comment:
This should be an `INFO` level logging. This is an important log message to debug the datastream state change. Also, `DatastreamGroup.toString()` method is an optimized method that prints only the datastream names, so it will be a small log message. You can check the size of the list and print only when it is non-zero.